### DISCUSSION TITLE
How to voluntarily exit 10000 validators?

### DISCUSSION THREAD

**Discussion by q9f** [ChainSafe/lodestar#6830] [Q&A] [answered]
GM GM

```
100% of local keystores imported. current=10000 total=10000 rate=114678.90keys/m
Loaded keystores via keystore cache
 ✖ Error: Can not fetch state validators from beacon node - Request Header Fields Too Large: Exceeded maximum allowed HTTP header size
    at Function.assert (file:///home/user/lodestar/packages/api/src/utils/client/httpClient.ts:44:13)
    at resolveValidatorIndexes (file:///home/user/lodestar/packages/cli/src/cmds/validator/voluntaryExit.ts:213:12)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at Object.handler (file:///home/user/lodestar/packages/cli/src/cmds/validator/voluntaryExit.ts:100:30)
```

I'm trying to exit my Holesky validators but it exceeds the http request header size. 

Is there a way I can specify the range or do I have to start collecting all public keys and queue them in batches?

**Comment by nflaig** [comment, marked as answer]
> I'm trying to exit my Holesky validators but it exceeds the http request header size.

We should probably switch to new post api to avoid this issue in the future, for now, you can just increase the header limit on the beacon node to allow the request by setting `--rest.headerLimit 1638400`

**Comment by nflaig** [comment]
> Is there a way I can specify the range or do I have to start collecting all public keys and queue them in batches?

We should probably batch internally to avoid the issue, right now you would have to specify the public keys explicitly but https://github.com/ChainSafe/lodestar/discussions/6830#discussioncomment-9606783 should work
